pkg/export/system.go

```diff
@@ -70,14 +70,16 @@ func (s *System) CloseConnection(connectionName string) error {
 }
 
 func (s *System) closeConnection(connectionName string) error {
-	if c, ok := s.connectionToDriver[connectionName]; ok {
-		if conn, ok := SupportedDrivers[c]; ok {
+	if driverName, ok := s.connectionToDriver[connectionName]; ok {
+		// connection should be deleted from the map before closing it to make sure old connection is not accessible if close fails
+		// also avoids not respecting the latest connection with the same name if close fails
+		delete(s.connectionToDriver, connectionName)
```

Comment:
what happens if closing a connection failed, is there any retry? with this change, if closing connection failed and it was deleted, we now have an unclosed connection that we no longer track. alternatively, we could track if closing succeeded or now in the map and only delete the connection after we know it has been closed successfully.

Reply:
you could also keep both connection and closed status in the same map, `defer delete` so it is always called regardless if close succeeded or not. in delete() always check if connect is closed. delete connect from map if connect is closed.